File: src-tauri/src/commands/secrets.rs
use crate::{context::ServiceContext, secret_store::KeyringSecretStore};
use std::sync::Arc;
use tauri::State;
use wealthfolio_core::secrets::SecretStore;

#[tauri::command]
pub async fn set_secret(
    secret_key: String,
    secret: String,
    _state: State<'_, Arc<ServiceContext>>, // keep signature consistent
) -> Result<(), String> {
    KeyringSecretStore
        .set_secret(&secret_key, &secret)
        .map_err(|e| e.to_string())
}

#[tauri::command]
pub async fn get_secret(
    secret_key: String,
    _state: State<'_, Arc<ServiceContext>>,
) -> Result<Option<String>, String> {
    KeyringSecretStore
        .get_secret(&secret_key)
        .map_err(|e| e.to_string())
}

#[tauri::command]
pub async fn delete_secret(
    secret_key: String,
    _state: State<'_, Arc<ServiceContext>>,
) -> Result<(), String> {
    KeyringSecretStore
        .delete_secret(&secret_key)
        .map_err(|e| e.to_string())
}

File: apps/frontend/src/adapters/shared/secrets.ts
// Secrets Commands
import { invoke } from "./platform";

export const setSecret = async (secretKey: string, secret: string): Promise<void> => {
  return invoke<void>("set_secret", { secretKey, secret });
};

export const getSecret = async (secretKey: string): Promise<string | null> => {
  return invoke<string | null>("get_secret", { secretKey });
};

export const deleteSecret = async (secretKey: string): Promise<void> => {
  return invoke<void>("delete_secret", { secretKey });
};
